modules/virt-adding-key-creating-vm-template.adoc

```diff
@@ -0,0 +1,77 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-accessing-vm-ssh.adoc
+
+ifeval::["{context}" == "static-key"]
+:static-key:
+:title: Adding a key
+endif::[]
+ifeval::["{context}" == "dynamic-key"]
+:dynamic-key:
+:title: Enabling dynamic key injection
+endif::[]
+
+:_content-type: PROCEDURE
+[id="virt-adding-key-creating-vm-template_{context}"]
+= {title} when creating a VM from a template
+
+ifdef::static-key[]
+You can add a statically managed public SSH key when you create a virtual machine (VM) by using the {product-title} web console. The key is added to the VM as a cloud-init data source at first boot. This method does not affect cloud-init user data.
+
+Optional: You can add a key to a project. Afterwards, this key is added automatically to VMs that you create in the project.
+endif::[]
+ifdef::dynamic-key[]
+You can enable dynamic public SSH key injection when you create a virtual machine (VM) from a template by using the {product-title} web console. Then, you can update the key at runtime.
+
+[NOTE]
+====
+Only {op-system-base-full} 9 supports dynamic key injection.
+====
+
+The key is added to the VM by the QEMU guest agent, which is installed with {op-system-base} 9.
+endif::[]
+
+.Prerequisites
+
+* You generated an SSH key pair by running the `ssh-keygen` command.
+
+.Procedure
+
+. Navigate to *Virtualization* -> *Catalog* in the web console.
+ifdef::dynamic-key[]
+. Click the *Red Hat Enterprise Linux 9 VM* tile.
+endif::[]
+ifdef::static-key[]
+. Click a template tile.
++
+The guest operating system must support configuration from a cloud-init data source.
```

Comment:
You can't make a prerequisite out of a requirement that cannot be fulfilled before the user starts the procedure. It's like saying this:

Prerequisite: You must open only red boxes. 
Procedure:
1. Open the closet.
2. Check the boxes.
3. Open the red box.

The user can't ensure that the OS supports cloud-init until they select a template. However, when they select a template, they must select an OS that supports cloud-init. I could have written it as a single sentence - select a template whose guest operating system supports cloud-init. However, that's rather verbose. Since most OSes DO support cloud-init, this is more of a reminder than a warning or command, just in case they have a custom template that does not support cloud-init.

No ".." before the line. It's attached to the step. It is not a sub-procedure.